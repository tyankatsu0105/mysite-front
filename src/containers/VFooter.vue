<template>
	<div class="VFooter">
		<div class="VFooter-Copyright">
			<small>&copy; {{ copyright }}</small>
		</div>
	</div>
</template>

<script>
import { SITE_NAME } from "@/const";
export default {
  data() {
    return {
      currentYear: new Date().getFullYear()
    };
  },
  computed: {
    copyright() {
      const createYear = 2018;

      let sinceYear;
      if (createYear === this.currentYear) {
        sinceYear = createYear;
      } else {
        sinceYear = `${createYear} - ${this.currentYear}`;
      }
      return `${sinceYear} ${SITE_NAME}`;
    }
  }
};
</script>

<style scoped lang='scss'>
.VFooter {
  width: 100%;
  padding: 12px 0;
  &-Copyright {
    text-align: center;
  }
}
</style>
