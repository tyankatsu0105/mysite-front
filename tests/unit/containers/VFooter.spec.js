import { shallowMount } from "@vue/test-utils";
import VFooter from "@/containers/VFooter.vue";

describe("VFooter", () => {
  it("render", () => {
    const wrapper = shallowMount(VFooter);
    wrapper.setData({
      currentYear: new Date("2018").getFullYear()
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
  describe("copyright", () => {
    let SITE_NAME;
    beforeEach(() => {
      SITE_NAME = "Tyankatsu Sketch Book";
    });
    it("2018", () => {
      const wrapper = shallowMount(VFooter);
      wrapper.setData({
        currentYear: new Date("2018").getFullYear()
      });
      expect(wrapper.vm.copyright).toEqual(`2018 ${SITE_NAME}`);
    });
    it("2018 - 2019", () => {
      const wrapper = shallowMount(VFooter);
      wrapper.setData({
        currentYear: new Date("2019").getFullYear()
      });
      expect(wrapper.vm.copyright).toEqual(`2018 - 2019 ${SITE_NAME}`);
    });
  });
});
