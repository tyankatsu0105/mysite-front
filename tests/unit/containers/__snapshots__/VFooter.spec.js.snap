// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VFooter render 1`] = `
<div class="VFooter">
  <div class="VFooter-Copyright"><small>Â© 2018 Tyankatsu Sketch Book</small></div>
</div>
`;
